packages/metadata/test/unit/factory/factory.spec.ts

```diff
@@ -0,0 +1,130 @@
+import * as fs from "fs";
+import { beforeEach, describe, expect, it, vi } from "vitest";
+
+import { Cache, ILogger } from "@zkchainhub/shared";
+
+import {
+    GithubMetadataProvider,
+    LocalFileMetadataProvider,
+    MetadataConfig,
+    MetadataProviderFactory,
+    MetadataSource,
+    StaticMetadataProvider,
+} from "../../../src/internal.js";
+import { mockChainData, mockTokenData } from "../../fixtures/metadata.fixtures.js";
+
+// Mock the file system functions
+vi.mock("fs", () => ({
+    existsSync: vi.fn(),
+    readFileSync: vi.fn(),
+}));
+
+describe("MetadataProviderFactory", () => {
+    let logger: ILogger;
+    let cache: Cache;
+
+    beforeEach(() => {
+        logger = {
+            info: vi.fn(),
+            warn: vi.fn(),
+            error: vi.fn(),
+            debug: vi.fn(),
+        };
+        cache = {
+            store: {} as any,
+            get: vi.fn(),
+            set: vi.fn(),
+            del: vi.fn(),
+            reset: vi.fn(),
+        };
+    });
+
+    describe("create", () => {
+        it("create a GithubMetadataProvider when metadata source is 'github' and dependencies are provided", () => {
+            const metadata: MetadataConfig<MetadataSource> = {
+                source: "github",
+                tokenUrl: "https://github.com/token",
+                chainUrl: "https://github.com/chain",
+            };
+            const deps = {
+                logger,
+                cache,
+            };
+
+            const provider = MetadataProviderFactory.create(metadata, deps);
+
+            expect(provider).toBeInstanceOf(GithubMetadataProvider);
+            expect(provider).toHaveProperty("tokenJsonUrl", "https://github.com/token");
+            expect(provider).toHaveProperty("chainJsonUrl", "https://github.com/chain");
+            expect(provider).toHaveProperty("logger", logger);
+            expect(provider).toHaveProperty("cache", cache);
+        });
+
+        it("throw an error when creating a GithubMetadataProvider and dependencies are missing", () => {
+            const metadata: MetadataConfig<MetadataSource> = {
+                source: "github",
+                tokenUrl: "https://github.com/token",
+                chainUrl: "https://github.com/chain",
```

Comment:
i would add a test for the case in which some or both of these two are missing

Reply:
i added validation on constructor and test cases but on the `GithubMetadataProvider` directly instead of the factory